### PR TITLE
Fix tao compiler error

### PR DESCRIPTION
**Pull request by lfengad**
- Fix shape utils error in disc compiler
- Disable ROCm disc-replay compilation for later support

### CODE REVIEW

**Review by minminsun — APPROVED** [review]
LGTM

File: tao_compiler/mlir/disc/transforms/shape_utils.cc [hunk 1]
```diff
@@ -656,7 +659,9 @@ SymbolDim* ShapeAnalysis::getRootDim(SymbolDim* symbolDim) {
 SymbolDim* ShapeAnalysis::getDim(Value value, int64_t dim) {
   auto it = shapeMap_.find(value);
   if (it == shapeMap_.end()) return nullptr;
-
+  if (it->second.rank() <= 0 || dim < 0 || dim > it->second.rank() - 1) {
```
Comment (wyzero) [inline tao_compiler/mlir/disc/transforms/shape_utils.cc:662]: add a file-check based UT?

File: scripts/python/tao_build.py [hunk 2]
```diff
@@ -495,7 +495,9 @@ def bazel_build(target, flag=""):
 
         bazel_build(TARGET_TAO_COMPILER_MAIN, flag=flag)
         bazel_build(TARGET_DISC_OPT, flag=flag)
-        bazel_build(TARGET_DISC_REPLAY, flag=flag)
+        # TODO:(fl237079) Support disc_replay for rocm version 
+        if not args.rocm and not args.dcu:
+            bazel_build(TARGET_DISC_REPLAY, flag=flag)
```
Comment (qiuxiafei) [inline scripts/python/tao_build.py:500]: What's the problem when building replay tool for rocm?

File: tao_compiler/mlir/disc/transforms/shape_utils.cc (hunk 1 above)
Comment (qiuxiafei) [reply tao_compiler/mlir/disc/transforms/shape_utils.cc:662]: +1

File: scripts/python/tao_build.py (hunk 2 above)
Comment (lfengad) [reply scripts/python/tao_build.py:500]: There are some GPU related library dependency problems.

File: tao_compiler/mlir/disc/transforms/shape_utils.cc (hunk 1 above)
Comment (lfengad) [reply tao_compiler/mlir/disc/transforms/shape_utils.cc:662]: added

Comment (minminsun) [reply tao_compiler/mlir/disc/transforms/shape_utils.cc:662]: Thanks. I tried this test out, and disc-opt will crash without the fix of this pr.

**Review by qiuxiafei — APPROVED** [review]

**Pull request merged**